presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -189,6 +192,41 @@ public void removeQuery(QueryId queryId)
         dynamicFilterContexts.remove(queryId);
     }
 
+    /**
+     * Dynamic filters are collected in same stage as the join operator. This can result in deadlock
+     * for source stage joins and connectors that wait for dynamic filters before generating splits
+     * (probe splits might be blocked on dynamic filters which require at least one probe task in order to be collected).
+     * To overcome this issue an initial task is created for source stages running broadcast join operator.
+     * This task allows for dynamic filters collection without any probe side splits being scheduled.
+     */
+    public boolean needsDynamicFiltersCollectingTask(QueryId queryId, PlanFragment plan)
```

Comment:
rename plan to fragment

Reply:
I find the name `needs...` to be a bit strange.  It sounds like the command `I need...`.  Consider renaming to `isCollectingTaskNeeded`.  Given this is the `DynamicFilterService` the `DynamicFilter` in the name seems redundant.